src/java.base/share/classes/java/lang/Class.java

```diff
@@ -185,6 +189,32 @@
  * a class or interface is hidden has no bearing on the characteristics
  * exposed by the methods of class {@code Class}.
  *
+ * <h2><a id=unnamedClasses>Unnamed Classes</a></h2>
+ *
+ * A {@code class} file representing an {@linkplain #isUnnamedClass unnamed class} is generated by
+ * a Java compiler from a source file for an unnamed class. The {@code
+ * Class} object representing an unnamed class is top-level,
+ * {@linkplain #isSynthetic synthetic}, and {@code final}. While an
+ * unnamed class does <em>not</em> have a name in its Java source
+ * form, several of the name-related methods of {@code java.lang.Class}
+ * do return non-null and non-empty results for the {@code Class}
+ * object representing an unnamed class.
+ *
+ * Conventionally, a Java compiler, starting from a source file for an
+ * unnamed class, say {@code HelloWorld.java}, creates a
+ * similarly-named {@code class} file, {@code HelloWorld.class}, where
+ * the class stored in that {@code class} file is named {@code
+ * "HelloWorld"}, matching the base names of the source and {@code
+ * class} files.
+ *
+ * For the {@code Class} object for an unnamed class {@code
+ * HelloWorld}, the methods to get the {@linkplain #getName name} and
+ * {@linkplain #getTypeName type name}
+ * return results equal to {@code "HelloWorld"} while the
+ * {@linkplain #getSimpleName
+ * simple name} of such an unnamed class is the empty string and the
+ * {@linkplain #getCanonicalName canonical name} is {@code null}.
```

Comment:
This sentence is quite long. You might to consider splitting it up, or at minimally put a comma after "empty string" to allow the reader draw breath.